cmake: Use correct installation dirs for doc and cmake configs

cryptomilk opened the pull request:
Fixes #623

Signed-off-by: Andreas Schneider <asn@cryptomilk.org>

cryptomilk committed:
cmake: Use correct installation dir for generated docs

Fixes #623

Signed-off-by: Andreas Schneider <asn@cryptomilk.org>

clanmills commented:
I'm going to ask @piponazo to review this as he's our CMake expert.  If we change the destination of the files required to "consume" CMake, we should make appropriate changes to README.md.

cryptomilk committed:
cmake: Use correct installation dir for cmake config files

This is the path cmake will look for those files by default. At least on
UNIX systems.

Fixes #623

Signed-off-by: Andreas Schneider <asn@cryptomilk.org>

piponazo commented on src/CMakeLists.txt:
```diff
@@ -234,7 +234,7 @@ install(FILES
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/exiv2)
 
-install(EXPORT exiv2Config DESTINATION "share/exiv2/cmake")
+install(EXPORT exiv2Config DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2")
```
I have tested this with the [exiv2Consumer](https://github.com/piponazo/exiv2Consumer) project and it works.

Just as a side note: the previous path was also correct. `find_package` try to search in different locations and both (the previous path and the new one work):
https://cmake.org/cmake/help/latest/command/find_package.html?highlight=find_package#search-procedure
